packages/shared/src/hooks/usePaddlePayment.ts

```diff
@@ -0,0 +1,192 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import type {
+  CheckoutCustomer,
+  CheckoutLineItem,
+  Environments,
+  Paddle,
+  PaddleEventData,
+} from '@paddle/paddle-js';
+import { CheckoutEventNames, initializePaddle } from '@paddle/paddle-js';
+import { useRouter } from 'next/router';
+import { useAuthContext } from '../contexts/AuthContext';
+import { useLogContext } from '../contexts/LogContext';
+import type { TargetType } from '../lib/log';
+import { LogEvent } from '../lib/log';
+import { plusSuccessUrl } from '../lib/constants';
+import { checkIsExtension } from '../lib/func';
+import type {
+  OpenCheckoutProps,
+  PaymentContextProviderProps,
+} from '../contexts/payment/context';
+
+interface UsePaddlePaymentProps
+  extends Pick<
+    PaymentContextProviderProps<PaddleEventData, CheckoutEventNames>,
+    'successCallback' | 'disabledEvents'
+  > {
+  targetType?: TargetType;
+  getProductQuantity?: (event: PaddleEventData) => number;
+}
+
+export const usePaddlePayment = ({
+  successCallback,
+  disabledEvents,
+  targetType,
+  getProductQuantity,
+}: UsePaddlePaymentProps) => {
+  const router = useRouter();
+  const { logEvent } = useLogContext();
+  const { user, geo } = useAuthContext();
+  const [paddle, setPaddle] = useState<Paddle>();
+  const isCheckoutOpenRef = useRef(false);
+  const logRef = useRef<typeof logEvent>();
+  logRef.current = logEvent;
+
+  const callbacks = useRef({
+    successCallback,
+    getProductQuantity,
+  });
+  callbacks.current = { successCallback, getProductQuantity };
```

Comment:
@dailydotdev/web-team thoughts on setting the callback props to a ref? This is so our `paddle` is not initialized unexpectedly every time these values changes, and that it is easy to miss these callback to be wrapped in a `useCallback`.

Reply:
Yes, but I would do each callback to different ref because `callbacks.current` would get different reference and might be stale depending on how its called.

Reply:
Cool. Let me have them separated. Thanks for the thoughts!